library/std/src/sys/windows/rand.rs

```diff
@@ -33,37 +30,30 @@ use crate::sys::c;
 /// [`HashMap`]: crate::collections::HashMap
 /// [`RandomState`]: crate::collections::hash_map::RandomState
 pub fn hashmap_random_keys() -> (u64, u64) {
-    Rng::open().and_then(|rng| rng.gen_random_keys()).unwrap_or_else(fallback_rng)
+    Rng::SYSTEM.gen_random_keys().unwrap_or_else(fallback_rng)
 }
 
-struct Rng(c::BCRYPT_ALG_HANDLE);
+struct Rng {
+    algorithm: c::BCRYPT_ALG_HANDLE,
+    flags: u32,
+}
 impl Rng {
-    #[cfg(miri)]
-    fn open() -> Result<Self, c::NTSTATUS> {
-        const BCRYPT_RNG_ALG_HANDLE: c::BCRYPT_ALG_HANDLE = ptr::invalid_mut(0x81);
-        let _ = (
-            c::BCryptOpenAlgorithmProvider,
-            c::BCryptCloseAlgorithmProvider,
-            c::BCRYPT_RNG_ALGORITHM,
-            c::STATUS_NOT_SUPPORTED,
-        );
-        Ok(Self(BCRYPT_RNG_ALG_HANDLE))
+    const SYSTEM: Self = Self::new(ptr::null_mut(), c::BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+
+    const fn new(algorithm: c::BCRYPT_ALG_HANDLE, flags: u32) -> Self {
```

Comment:
Arguably should be an `unsafe fn` since we know at least WINE might read through the pointer. Doesn't matter much though, and I'm not sure what the precise contract would be beyond "don't mess the args up".

Reply:
Oh, that's a fair point.